### ISSUE TITLE
AppSettings string value is changed by Set-AzWebApp

### ISSUE THREAD

**Issue by skurik** [Azure/azure-powershell#10905]
## Description

I am using __Set-AzWebApp__ to set app settings on a web app I create dynamically (using __New-AzWebApp__). One of the settings is the web app creation time (as I did not find any other way how to get this information using Azure PowerShell). I am using the neutral round-trip format of the datetime to be able to reliably parse the string later. However, even though the string looks as expected locally, on the web app it is in a different format:

```
On the DevOps agent
2020-01-15T16:43:58.2385875Z 

In the web app AppSetting
01/15/2020 16:45:06 
```

## Steps to reproduce

```powershell

$AppSettings = @{
    "WEBSITE_SITE_TYPE" = "live"    
    "CREATED_UTC" = [DateTime]::UtcNow.ToString("o", [CultureInfo]::InvariantCulture)
}

Write-Host (ConvertTo-Json $AppSettings)

Set-AzWebApp -ResourceGroupName $TargetResourceGroup -Name $TargetAppName -AppSettings $AppSettings
```

This outputs 

```
{
    "WEBSITE_SITE_TYPE":  "live",
    "CREATED_UTC":  "2020-01-15T16:43:58.0823349Z"
}
```

as expected. However, after this runs, the web app shows

![image](https://user-images.githubusercontent.com/951352/72454762-c87be500-37c1-11ea-8a6e-62023dcf7946.png)

## Module versions

```powershell
=================
Task         : Azure PowerShell
Description  : Run a PowerShell script within an Azure environment
Version      : 4.159.7
Author       : Microsoft Corporation
Help         : [Learn more about this task](https://go.microsoft.com/fwlink/?LinkID=613749)
=================

VERBOSE: Loading module from path 
'C:\Modules\az_3.1.0\Az.Websites\1.5.0\Microsoft.Azure.PowerShell.Cmdlets.Websites.dll'
```

**Comment by panchagnula** [comment]
What is the output from the Set-AzWebApp command? 
is this the one you are referring to, which you mentioned above as correct?
{
    "WEBSITE_SITE_TYPE":  "live",
    "CREATED_UTC":  "2020-01-15T16:43:58.0823349Z"
}

If so I am wondering the bug is portal display? let me know. Thanks!

**Comment by skurik** [comment]
@panchagnula I updated the question. The output I quoted was that of __Write-Host__, proving that the  __$AppSettings__ variable contains correct data.

If I modify the __Set-AzWebApp__ call to

```
$Result = Set-AzWebApp -ResourceGroupName $TargetResourceGroup -Name $TargetAppName -AppSettings $AppSettings
Write-Host (ConvertTo-Json $Result.SiteConfig.AppSettings)
```

I get

```
[
    {
        "Name":  "WEBSITE_SITE_TYPE",
        "Value":  "live"
    },    
    {
        "Name":  "CREATED_UTC",
        "Value":  "01/15/2020 20:50:56"
    }
]
```

ie. an incorrect value.

What I am most intrigued by is that it is a _string_ value in the __$AppSettings__ hashtable, so how is it possible that it is apparently being interpreted as a DateTime at some point and serialized again.

**Comment by panchagnula** [comment]
thanks for the update @skurik , i think the format of the date time if the string is in this format the lack of Z in the end,  Let me try reproing this on our end. Thanks!

**Comment by skurik** [comment]
@panchagnula Just a wild guess, but I can imagine there is some JSON parser involved when applying the app settings sent by __Set-AzWebApp__, right? In that case, it could be the following: https://github.com/JamesNK/Newtonsoft.Json/issues/862

In other words

```
JsonConvert.DeserializeObject<dynamic>(
    JsonConvert.SerializeObject(new { Prop = "2020-01-15T16:43:58.0823349Z" })
).Prop == "1/15/2020 16:43:58"
```

Because the JSON parser (Json.NET in this case) is "smart" enough to recognize the string looks like a DateTime, parses it to a DateTime and then, well, I have no control over how it serializes it when asked to.

**Comment by ThejaChoudary** [comment]
Taking  a look at this.

**Comment by panchagnula** [comment]
@skurik there is some confusion on our end in understanding the ask here. Are you saying that the format of the date changed or is it the time that is modified by PowerShell command?
Are you expecting  the return value to Jan 15th 2020 and the US format on the portal for this in the form MONTH/DATE/YEAR incorrect as per you or is the issue here regarding the time changing from 16:45 to 20:50? Please clarify so that we can proceed with the investigation/fix. Thanks!

**Comment by skurik** [comment]
@panchagnula The problem is that when I do the following

```
$AppSettings = @{    
    CREATED_UTC = "2020-01-15T16:43:58.0823349Z"
}

Set-AzWebApp -AppSettings $AppSettings <other params>
```

then the actual app settings on the target site (in Azure portal or being retrieved by a Az module) show that there is a __CREATED_UTC__ key with a value __01/15/2020 20:50:56__

In another words, I am setting a __string__ value which for some reason, some component on the way (__Set-AzWebApp__, the App Service engine or whatever) interprets as a __date time__ and then re-serialize it __in another format__.

In yet another words, the expected behavior is that whatever I use as a value, it is not touched or interpreted in any way and is stored __as I specified it__. 

Here, by coincidence, the string value _is_ a date time, but Azure __must__ treat it as an opaque value and not try to interpret it.

**Comment by panchagnula** [comment]
ok if its a normal string & not Datetime : the value returned would be identical to the value set. I am not sure i agree that DateTime format should be treated like an opaque string. what happens if you set the value as a string "2020-01-15T16:43:58.0823349" ?

**Comment by skurik** [comment]
@panchagnula 

App settings is a key/value map where keys are strings and values are strings. It is __not__ a set of typed values.

I need to be able use any string value without Azure interfering and interpreting it. It is the reasonable, expected and intuitive behavior. 

I consider the current behavior a bug and I don't understand why I should be forced to mangle the value in order to have it stored as it is.

**Comment by panchagnula** [comment]
fair point - I guess I like API being smart at detecting the type of data especially since AppSettings can be used as a key-value pair & used for anything. Given this is not specific to PowerShell I am going to move it to the API repo & also give a heads up to the API team. Thanks for your patience in dealing with this.
